### PR TITLE
Color tokens extraction

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

1. Added the necessary logic to extract the Colors from our Figma library.
2. Performed an initial extraction of the color tokens
3. Built a demo page on the plasma website

**Review might be easier if you review commit by commit**. A lot of new files are just the result of the extraction so no need to really review those.

### Potential Breaking Changes

None.

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by guscoveo** [comment]
Hey Gaël, glad the whole is taking shape! I would just have two minor comments:
- Any chances that in Plasma we can list the colors by order (as in Figma)? So first go the primary, then accents, then secondary. Just a detail but it will be much less confusing for newcomers.
- I was gonna say something about the shadow, but I'll just drop it... heh

**Comment by gdostie** [comment]
> Any chances that in Plasma we can list the colors by order (as in Figma)? So first go the primary, then accents, then secondary.

The order figma gives me from its API is not the one you would expect. First color is gradient, then Lime, then Indigo... I feel its all mixed up, so I went with alphabetical order which is less confusing to new comers

**Review by GermainBergeron — APPROVED** [review]

File: packages/website/src/pages/foundations/Colors.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+const ColorGroup: FunctionComponent<{name: string; value: any}> = ({name, value}) => {
+    if (!isColorGroup(value)) {
+        return <ColorBlock colorName={name} colorValue={value} />;
+    }
+
+    return (
+        <>
+            {Object.keys(value)
+                .sort()
```
[the remark this answers is not in the document]
Comment (gdostie) [reply packages/website/src/pages/foundations/Colors.tsx:47]: Yes, it sorts everything alphabetically

File: packages/tokens/README.md [hunk 2]
```diff
@@ -24,23 +24,46 @@ Tokens are provided through multiple outputs. Choose the format that best suits
...
 
-Coming soon.
+```scss
+@import '~@coveord/plasma-tokens/scss';
+// or individual libraries
+@import '~@coveord/plasma-tokens/scss/Colors';
 
-### Icons
+.something {
+    background-color: $color-primary-action-blue-6;
```
[the remark this answers is not in the document]
Comment (gdostie) [reply packages/tokens/README.md:41]: Good idea, I will make the change

**Comment by aiheon** [comment]
I was wondering does the token extraction now extract colors AND icons at once? or you can pick which one you want? 🤔

**Review by aiheon — APPROVED** [review]

**Comment by gdostie** [comment]
> I was wondering does the token extraction now extract colors AND icons at once? or you can pick which one you want? 🤔

Both, you can pick only the libraries you want and all of them if you need to.

File: packages/website/src/pages/foundations/Colors.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (GermainBergeron) [reply packages/website/src/pages/foundations/Colors.tsx:47]: The following sort just re-order the group of the same level?

**Comment by gdostie** [comment]
I made a small hack to make the primary colors appear first, then the rest is sorted alphabetically like before.

**Pull request merged**